python/flexflow/core/flexflow_cffi.py

```diff
@@ -15,19 +15,20 @@
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import cffi
-import os
-import subprocess
-import logging
 import warnings
 import numpy as np
 from .flexflow_logger import fflogger
 from flexflow.type import ActiMode, RegularizerMode, AggrMode, PoolType, DataType, LossType, CompMode, MetricsType, InferenceMode, ModelType, OpType, ParameterSyncType, enum_to_int, int_to_enum
+from flexflow.config import *
+from .flexflowlib import ffi, flexflow_library
 
-_FF_BUILD_DOCS = bool(os.environ.get('READTHEDOCS') or os.environ.get("FF_BUILD_DOCS"))
-if not _FF_BUILD_DOCS:
-  from .flexflowlib import ffi, flexflow_library
+def ffc():
+  if not flexflow_already_initialized():
+    raise RuntimeError("Cannot use FlexFlow library before initializing FlexFlow")
   ffc = flexflow_library.lib
+  if ffc is None:
```

Comment:
why ffc could be None?

Reply:
It shouldn't be None, this is just to catch any potential bug where flexflow is marked as initialized, but the library was not actually initialized in time for the first command that uses it